app/plugin_key_value_store.go

```diff
@@ -81,17 +98,17 @@ func (a *App) CompareAndDeletePluginKey(pluginId string, key string, oldValue []
 func (a *App) GetPluginKey(pluginId string, key string) ([]byte, *model.AppError) {
 	if kv, err := a.Srv().Store.Plugin().Get(pluginId, key); err == nil {
 		return kv.Value, nil
-	} else if err.StatusCode != http.StatusNotFound {
+	} else if _, ok := err.(*store.ErrNotFound); !ok {
 		mlog.Error("Failed to query plugin key value", mlog.String("plugin_id", pluginId), mlog.String("key", key), mlog.Err(err))
-		return nil, err
+		return nil, model.NewAppError("GetPluginKey", "app.plugin_store.get.app_error", nil, err.Error(), http.StatusInternalServerError)
```

Comment:
This should return a 404, not 500.

Reply:
I don't agree, in fact here we are checking for an error distinct of`ErrNotFound`

Reply:
Sorry, could you clarify? Yes we are checking for ErrNotFound, and that's why I would like to return 404. The original code also does the same.